proposals/redirects.md

```diff
@@ -0,0 +1,116 @@
+<!--
+  Note: You are probably looking for `stage-1--discussion-template.md`!
+  This template is reserved for anyone championing an already-approved proposal.
+
+  Community members who would like to propose an idea or feature should begin
+  by creating a GitHub Discussion. See the repo README.md for more info.
+
+  To use this template: create a new, empty file in the repo under `proposals/${ID}.md`.
+  Replace `${ID}` with the official accepted proposal ID, found in the GitHub Issue
+  of the accepted proposal.
+-->
+
+- Start Date: 2023-05-22
+- Reference Issues: https://github.com/withastro/roadmap/issues/466
+- Implementation PR: https://github.com/withastro/astro/pull/7067/
+
+# Summary
+
+Add a `redirects` config option to the Astro config which allows you to define redirects in a central place. Allow integrations to read this information and apply it to it's own redirects configuration for deployment.
+
+# Example
+
+New config option used like so:
+
+```js
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  redirects: {
+    '/other': '/place'
+  }
+});
+```
+
+# Background & Motivation
+
+This was original proposed as a stage 1 discussion [here](https://github.com/withastro/roadmap/discussions/319) and was one of the top voted proposals.
+
+As websites age there are times where routes are rearchitectured. In order preserve existing links to content on the web it is common to set up redirects in your web server.
+
+Redirect configuration varies depending on what web server or host you use. Some times you might even want to change hosts, in which case your redirects need to be converted over to a new format.
+
+Having a redirects configuration within Astro itself allows a single place to define redirects that works everywhere.
+# Goals
+
+- Works in development mode.
+- Writes out `<meta http-equiv="refresh">` tags in static builds.
+- Gives proper hooks to integrations so that they can write to their own configuration.
+
+# Non-Goals
+
+- Specifying temporary redirects. These are best done at runtime, so SSR is a more appropriate way to do those.
+    - In the future we could expand this API to give more control over what status code is used, once use-cases are known.
+
+# Detailed Design
+
+This will be implemented as a feature of the internal routing. The `RouteData` type will be extended to include:
+
+```js
+export interface RouteData {
+  type: 'redirect';
+  // ...
+  redirect?: string;
+}
+```
+
+Our core rendering handles routing and will detect this type of route and return a `Response` with a status code of `301` with the `Location` header set to the value of the route's `redirect` property.
```

Comment:
What about API endpoints? Will they be treated as 301 too? Or should we use [308](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/308) to keep the method used?